### PR TITLE
Initial implementation of I2CDisplay for Blinka

### PR DESCRIPTION
**Pull request by lesamouraipourpre**
This PR collects the suggestions of @eteq and @wolfmanjm from PR #41. 

I have (minimally) tested on a Raspberry Pi 3 and it works. It could do with being tested on other Blinka devices.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -25,7 +25,13 @@
...
 # digitalio, micropython and busio. List the modules you use. Without it, the
 # autodoc module docs will fail to generate with a warning.
-# autodoc_mock_imports = ["digitalio", "busio"]
+autodoc_mock_imports = [
+    "busio",
+    "digitalio",
+    "displayio",
+    "microcontroller",
+    "terminalio",
+]
```
Comment (makermelissa) [inline docs/conf.py:34]: Is there a reason you are using autodoc_mock_imports? I think all of these are already in requirements, so there's no need to mock the modules up.

**Review by makermelissa — CHANGES_REQUESTED** [review]
Thanks for submitting this. This looks great other than the autodoc_mock_imports.

File: docs/conf.py (hunk 1 above)
Comment (lesamouraipourpre) [reply docs/conf.py]: I've trimmed the list down in the most recent push, but if I don't include `displayio` and `terminalio` the docs won't build.

Comment (makermelissa) [reply docs/conf.py]: That's odd considering both `displayio` and `terminalio` are part of this repo.

Comment (lesamouraipourpre) [reply docs/conf.py]: Give me a while to investigate further. I've just realised that the generated docs are wrong.

**Comment by lesamouraipourpre** [comment]
The `autodoc_mock_imports` line was unnecessary. Somehow my virtual environment had become corrupted.
Tested by creating a new virtual environment and re-cloning my repo.

**Comment by makermelissa** [comment]
Ok, once https://github.com/adafruit/Adafruit_Blinka/pull/488 is merged in and a newer version of Blinka is released, we can re-run this and the checks should pass.

**Comment by makermelissa** [comment]
It's possible that it may not like microcontroller.Pin anyways since it is just passing, so I may need to create a generic pin class for sphinx. In that case, adding `microcontroller` to autodoc_mock_imports makes sense.

**Review by makermelissa — APPROVED** [review]

**Pull request merged**